libsolidity/ast/AST.h

```diff
@@ -1722,6 +1722,28 @@ class Throw: public Statement
 	void accept(ASTConstVisitor& _visitor) const override;
 };
 
+/**
+ * The revert statement is used to revert state changes and return error data.
+ */
+class RevertStatement: public Statement
+{
+public:
+	explicit RevertStatement(
+		int64_t _id,
+		SourceLocation const& _location,
+		ASTPointer<ASTString> const& _docString,
+		ASTPointer<FunctionCall> _functionCall
+	):
+		Statement(_id, _location, _docString), m_errorCall(std::move(_functionCall)) {}
```

Comment:
What about adding assertion?
```suggestion
		Statement(_id, _location, _docString), m_errorCall(std::move(_functionCall))
	{
		solAssert(m_errorCall, "");
	}
```

Reply:
We almost never do that in the AST, but I can add it.